journal.py

```diff
@@ -134,37 +139,53 @@ def read_entries(request):
 
 
 # EDIT ENTRY PAGE
-@view_config(route_name='edit', renderer='templates/edit.jinja2')
+@view_config(route_name='edit', renderer='json')
 def edit(request):
     """view for edit single entry"""
+    entry = ""
+    id = request.params.get('id', None)
     if request.authenticated_userid:
         if request.method == 'POST':
             try:
-                id = request.matchdict.get('id', -1)
                 update_entry(request, id)
+                entry = get_single_entry(request, True, True, id)
             except psycopg2.Error:
                 # this will catch any errors generated by the database
                 return HTTPInternalServerError
-            return HTTPFound(request.route_url('home'))
-    return get_single_entry(request)
+        else: # GET REQUEST
+            entry = get_single_entry(request, False, True, id)
+    entry['created'] = entry['created'].strftime('%b %d, %Y')
+    return entry
+
 
 
 # DETAIL PAGE
 @view_config(route_name='detail', renderer='templates/detail.jinja2')
 def detail_view(request):
     """view for entry (permalink for an entry)."""
-    return get_single_entry(request, True)
+    id = request.matchdict.get('id', -1)
+    return get_single_entry(request, True, False, id)
 
+def get_newest_entry(request):
+    cursor = request.db.cursor()
+    cursor.execute(SELECT_SINGLE_NEWEST_ENTRY)
+    keys = ('id', 'title', 'text', 'created')
+    entry = dict(zip(keys, cursor.fetchone()))
+    entry['text'] = mdown(entry['text'])
+    entry['created'] = entry['created'].strftime('%b %d, %Y')
+    return entry
 
-def get_single_entry(request, mark_down=False):
+def get_single_entry(request, mark_down, json, id):
```

Comment:
Nice helper function here too.  These really help clean up the rest of your code.  Well factored!